swap/htlc.go

```diff
@@ -650,44 +630,50 @@ func (h *HtlcScriptV2) lockingConditions(htlcOutputType HtlcOutputType,
 
 // HtlcScriptV3 encapsulates the htlc v3 script.
 type HtlcScriptV3 struct {
-	// The final locking script for the timeout path which is available to
-	// the sender after the set blockheight.
+	// TimeoutScript is the final locking script for the timeout path which
+	// is available to the sender after the set blockheight.
 	TimeoutScript []byte
 
-	// The final locking script for the success path in which the receiver
-	// reveals the preimage.
+	// SuccessScript is the final locking script for the success path in
+	// which the receiver reveals the preimage.
 	SuccessScript []byte
 
-	// The public key for the keyspend path which bypasses the above two
-	// locking scripts.
+	// InternalPubKey is the public key for the keyspend path which bypasses
+	// the above two locking scripts.
 	InternalPubKey *btcec.PublicKey
 
-	// The taproot public key which is created with the above 3 inputs.
+	// TaprootKey is the taproot public key which is created with the above
+	// 3 inputs.
 	TaprootKey *btcec.PublicKey
+
+	// RootHash is the root hash of the taptree.
+	RootHash chainhash.Hash
 }
 
 // newHTLCScriptV3 constructs a HtlcScipt with the HTLC V3 taproot script.
-func newHTLCScriptV3(cltvExpiry int32, senderHtlcKey,
-	receiverHtlcKey [33]byte, sharedKey *btcec.PublicKey,
+func newHTLCScriptV3(cltvExpiry int32, senderHtlcKey, receiverHtlcKey [33]byte,
 	swapHash lntypes.Hash) (*HtlcScriptV3, error) {
 
-	receiverPubKey, err := btcec.ParsePubKey(
-		receiverHtlcKey[:],
-	)
+	senderPubKey, err := schnorr.ParsePubKey(senderHtlcKey[1:])
+	if err != nil {
+		return nil, err
+	}
+
+	receiverPubKey, err := schnorr.ParsePubKey(receiverHtlcKey[1:])
 	if err != nil {
 		return nil, err
 	}
 
-	senderPubKey, err := btcec.ParsePubKey(
-		senderHtlcKey[:],
+	aggregateKey, _, _, err := musig2.AggregateKeys(
+		[]*btcec.PublicKey{senderPubKey, receiverPubKey}, true,
 	)
 	if err != nil {
 		return nil, err
 	}
 
 	var schnorrSenderKey, schnorrReceiverKey [32]byte
-	copy(schnorrSenderKey[:], schnorr.SerializePubKey(senderPubKey))
-	copy(schnorrReceiverKey[:], schnorr.SerializePubKey(receiverPubKey))
+	copy(schnorrSenderKey[:], senderHtlcKey[1:])
```

Comment:
What do you think about changing the signature of `GenSuccessPathScript` and `GenTimeoutPathScript` to accept the parsed `btcec.PublicKey` instead and then use `schnorr.SerializePubKey()` within those functions to make it a bit more explicit that we're always using the Schnorr x-only public key format in those scripts?